### PR TITLE
Reader: Builds endpoints preserving desired version.

### LINKED ISSUE #5189: Reader: Topic service should not use `WordPressRestApiEndpointURL` for path construction
Using the `WordPressRestApiEndpointURL` const is problematic if it should later be updated. Also, it looks like its pointing at v1.1 and our preferred API version for the reader is v1.2.   Update this for better versioning. 

The specific code is: 

```
        if (isFeed) {
            topic.path = [NSString stringWithFormat:@"%@read/feed/%@/posts/", WordPressRestApiEndpointURL, siteInfo.feedID];
        } else {
            topic.path = [NSString stringWithFormat:@"%@read/sites/%@/posts/", WordPressRestApiEndpointURL, siteInfo.siteID];
        }
```

### PR DESCRIPTION
**Pull request by aerych**
Fixes #5189 

To test:
Follow a feed (mashable, arstechnica) and a site. 
In your reader's following list, tap on the site name in a post card of both the feed and a site. 
Confirm that the feed and site loads correctly. 

Needs review: @koke

### CODE REVIEW

**Comment by koke** [comment]
I could have just said "I'd rather store relative paths".

But I started to follow the code and ended up implementing it:

https://gist.github.com/koke/1b30d861e6bb9d366f69bc186d0e9525

It seems to work properly, although I'm not sure if the stored "Current Topic" will have some trouble, or if it's worth migrating. Or if I missed something else completely, you definitely know the reader code better.

I even wrote a blog post on [creating and applying git patches](https://koke.me/2016/04/26/attaching-patches-to-pull-requests/) 😉

**Comment by aerych** [comment]
> I could have just said "I'd rather store relative paths".

Nice patch @koke :)  But I want to continue storing the full path, not relative paths. Two reasons:
1. fetched content should be version consistent. If we fetch tags from the v1.2 endpoint, their posts should be fetched from the v1.2 endpoint. This is easy enough to do with relative pathing but we get it for free by storing the full path. 
2. we don't today, but we potentially could consume content from different versions. A hypothetical example would be enhanced Discover content on a 1.4 or 2.0 endpoint, along with matching posts objects, while still consuming tags and sites on a 1.2 endpoint. Relative pathing would make this kind of arrangement troublesome.

**Comment by koke** [comment]
Hmm, I could argue against that but I'll leave it up to you, since you definitely know more about the reader current implementation and future plans.

However I'd ask you not to expose `WordPressComApiClientEndpointURL`. I was recently trying to extract WordPress API code into a separate framework inside the project (#5187) and had to postpone it because of all the coupling. I think exposing this constant would make it worse.

What if instead of the full URL you stored the absolute path (including the `/rest/v1.x`)?

**Comment by aerych** [comment]
> What if instead of the full URL you stored the absolute path (including the /rest/v1.x)?

That would probably work. I'll take a peek this afternoon.

**Comment by aerych** [comment]
I went a different direction.  It really makes the most sense to compose the endpoint path at the network level rather than the service level. Bonus, we can avoid referencing any constants and use the existing methods for composing a versioned path.  Ready for another round @koke!

File: WordPress/Classes/Networking/RemoteReaderSiteInfo.h [hunk 1]
```diff
@@ -14,5 +14,6 @@
 @property (nonatomic, strong) NSString *siteName;
 @property (nonatomic, strong) NSString *siteURL;
 @property (nonatomic, strong) NSNumber *subscriberCount;
+@property (nonatomic, strong) NSString *path;
```
Comment (koke) [inline WordPress/Classes/Networking/RemoteReaderSiteInfo.h]: In the context of SiteInfo, `path` looks a bit ambiguous to me. Maybe something a bit more descriptive like `postsEndpoint`?

File: WordPress/Classes/Networking/ReaderTopicServiceRemote.m [hunk 2]
```diff
@@ -234,9 +236,18 @@ - (RemoteReaderSiteInfo *)siteInfoForFeedResponse:(NSDictionary *)response
     if (![siteInfo.siteName length] && [siteInfo.siteURL length] > 0) {
         siteInfo.siteName = [[NSURL URLWithString:siteInfo.siteURL] host];
     }
+    NSString *endpointPath = [NSString stringWithFormat:@"read/feed/%@/posts/", siteInfo.feedID];
+    siteInfo.path = [self endointForPath:endpointPath];
     return siteInfo;
 }
 
+- (NSString *)endointForPath:(NSString *)endpoint
```
Comment (koke) [inline WordPress/Classes/Networking/ReaderTopicServiceRemote.m]: `endoint` 😉

File: WordPress/Classes/Networking/ReaderTopicServiceRemote.m [hunk 3]
```diff
@@ -234,9 +236,18 @@ - (RemoteReaderSiteInfo *)siteInfoForFeedResponse:(NSDictionary *)response
...
+{
+    NSString *path = [self pathForEndpoint:endpoint withVersion:ServiceRemoteRESTApiVersion_1_2];
```
Comment (koke) [inline WordPress/Classes/Networking/ReaderTopicServiceRemote.m]: Pretty minor, but the naming around this is a bit weird: there's a `...ForPath` parameter named `endpoint`, which it's then turned into a `path`. How about:

```objc
- (NSString *)endpointUrlForPath:(NSString *)path
{
    NSString *absolutePath = [self pathForEndpoint:path withVersion:ServiceRemoteRESTApiVersion_1_2];
    NSURL *url = [NSURL URLWithString:absolutePath relativeToURL:self.api.baseURL];
    return [url absoluteString];
}
```

**Comment by koke** [comment]
I really like the new approach more. Left a few minor comments around naming, but :shipit: otherwise.
Travis is failing because of the Gridicons pod, can you try merging develop first and see if it's green?

**Comment by koke** [comment]
BTW, was this really meant for 6.2? It didn't seem like an urgent fix

**Comment by aerych** [comment]
> Left a few minor comments around naming

Naming is hard :) I like your suggestions so let's do that.

**Comment by aerych** [comment]
> BTW, was this really meant for 6.2? It didn't seem like an urgent fix

It is, actually, tho its really a minor fix. Posts should be fetched from the v1.2 endpoints so we're sure to get all the things and the old code was fetching from v1.1.

**Comment by aerych** [comment]
Thanks @koke !!

**Pull request merged**